kafka-connect-http/src/main/scala/io/lenses/streamreactor/connect/http/sink/HttpSinkTask.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.http.sink
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import cats.implicits._
+import io.lenses.streamreactor.common.utils.JarManifest
+import io.lenses.streamreactor.connect.http.sink.config.HttpSinkConfig.fromJson
+import io.lenses.streamreactor.connect.http.sink.config.HttpSinkConfigDef.configProp
+import io.lenses.streamreactor.connect.http.sink.client.HttpRequestSender
+import io.lenses.streamreactor.connect.http.sink.config.HttpSinkConfig
+import io.lenses.streamreactor.connect.http.sink.tpl.templates.ProcessedTemplate
+import io.lenses.streamreactor.connect.http.sink.tpl.templates.RawTemplate
+import io.lenses.streamreactor.connect.http.sink.tpl.templates.Template
+import org.apache.kafka.connect.sink.SinkRecord
+import org.apache.kafka.connect.sink.SinkTask
+import org.http4s.jdkhttpclient.JdkHttpClient
+
+import java.util
+import scala.jdk.CollectionConverters.IterableHasAsScala
+import scala.jdk.CollectionConverters.MapHasAsScala
+
+class HttpSinkTask extends SinkTask {
+  private val manifest = JarManifest(getClass.getProtectionDomain.getCodeSource.getLocation)
+
+  private var maybeConfig: Option[HttpSinkConfig] = Option.empty
+
+  private var maybeTemplate: Option[Template] = Option.empty
+  override def start(props: util.Map[String, String]): Unit = {
+    {
+      for {
+        propVal <- props.asScala.get(configProp).toRight(new IllegalArgumentException("No prop found"))
+        config  <- fromJson(propVal)
+        template = RawTemplate(config.endpoint, config.content, config.headers)
+          .parse()
+      } yield {
+        this.maybeConfig   = config.some
+        this.maybeTemplate = template.some
+      }
+    }.leftMap(throw _)
+    ()
+
+  }
+
+  override def put(records: util.Collection[SinkRecord]): Unit = {
+    val sinkRecords = records.asScala
+    (maybeConfig, maybeTemplate) match {
+      case (Some(config), Some(template)) =>
+        val sender         = new HttpRequestSender(config)
+        val clientResource = JdkHttpClient.simple[IO]
+
+        clientResource.use {
```

Comment:
If worthwhile we can avoid allocating a client per put call.
A way that could be done would be to use the `.allocate` method on resource, something like what's done [here](https://github.com/lensesio-dev/lenses-core/blob/feat/RB-202_CE_3_Upgrade/flows-applications-http/src/test/scala/io/lenses/flows/application/util/DispatcherProvider.scala)